src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -623,14 +704,40 @@ public bool HasSkipLocalsInitAttribute
         internal bool IsAutoPropertyWithGetAccessor
             => IsAutoProperty && _getMethod is object;
 
-        protected bool IsAutoProperty
-            => (_propertyFlags & Flags.IsAutoProperty) != 0;
+        internal bool IsAutoProperty
```

Comment:
>IsAutoProperty [](http://example.com/codeflow?start=22&length=14)

Could you add test cases for this error scenario:
```
class Test
{
    int X
    {
        get
        {
            field = 3;
            X = 3;
            return 0;
        }
    }
}
```

And for this success scenario:
```
class Test
{
    Test()
    {
        X = 3;
        Console.WriteLine(X);
    }
    
    int X
    {
        get {return field;}
    }
}
```

Reply:
Interesting. It looks like binding needs to know whether it's an auto property in `AccessingAutoPropertyFromConstructor`, but `IsAutoProperty` needs binding, and it's a cycle again.

I think we'd like `IsAutoProperty` to not cause binding. But I can't think of any way to do that.

A possible fix is to introduce a binding flag that does a minimal amount of binding to determine whether we have a field keyword. But I'm not sure if this is a achievable and/or is good solution.

@AlekseyTs What do you think?